pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -423,8 +424,18 @@ func (p *provider) getConfig(s v1alpha1.ProviderSpec) (*Config, *providerconfigt
 			return nil, nil, nil, errors.New("Invalid value for `diskIops` (min: 100, max: 64000)")
 		}
 
+		c.DiskIops = rawConfig.DiskIops
+	} else if c.DiskType == ec2.VolumeTypeGp3 && rawConfig.DiskIops != nil {
+		// gp3 disks start with 3000 IOPS by default, we _can_ pass better IOPS, but it is not a required field
+		iops := *rawConfig.DiskIops
```

Comment:
Can we please move the null check for `rawConfig.DiskIops`  out of the if blocks altogether to remove redundancy?  
What you are doing is right too but I think instead of bypassing it with `else if c.DiskType == ec2.VolumeTypeGp3 && rawConfig.DiskIops != nil ` we should return an **explicit** error if it's nil since it's a **required field**. 

Similar to https://github.com/kubermatic/machine-controller/blob/184dd4bdde4d6378eb6979bcb94e4049afcc19ef/pkg/cloudprovider/provider/aws/provider.go#L419

Maybe something like this should do: 

```
if len(c.DiskType) > 0 && rawConfig.DiskIops == nil {
	return nil, nil, nil, errors.New("Missing required field `diskIops`")
}
```

Reply:
It's actually not a required field for `gp3`, that's why I added the comment. See "Iops" on https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVolume.html#API_CreateVolume_RequestParameters. Setting IOPS when creating a volume is only required for `io1` (and `io2`) disks.

So, for `gp3` disks, we only want to validate the field if it's set. If it's not, there's nothing to validate, and you get a `gp3` disk with 3000 IOPS.

Reply:
Got it, that makes sense :+1: